### DISCUSSION TITLE
Do not see logs emitted from from library

### DISCUSSION THREAD

**Discussion by siddhsql** [gabime/spdlog#3175] [Q&A]
I am trying to use `spdlog` like this:

```
auto sink = std::make_shared<spdlog::sinks::win_eventlog_sink_mt>(EVENT_SOURCE, MSG_GET_LUCKY);
        sink->set_pattern("%v");
        auto logger = std::make_shared<spdlog::logger>("eventlog", sink);        
        logger->set_level(spdlog::level::trace); 
        spdlog::set_default_logger(logger);
```

With this I do see messages in windows event log that are emitted from my code. However, I am not seeing any messages emitted by the libraries that my application is using. E.g., if there is a statement like this in a library that my application is using:

```
spdlog::debug("Hello World!");
```

I do not see it in the event log. What gives? How can I fix this?

Debugging the code I found that in the library code `spdlog` has level set to 2 (Info) mysteriously and it seems to be using a different sink as well:

![image](https://github.com/user-attachments/assets/939209ab-6a77-4fce-aa45-3d37a88487fb)

this seems like a bug to me.

**Comment by siddjain** [comment]
https://github.com/gabime/spdlog/issues/1598
https://github.com/gabime/spdlog/wiki/How-to-use-spdlog-in-DLLs
